cfn-resources/network-container/cmd/resource/delete.go

```diff
@@ -65,22 +67,41 @@ func deleteOperation(req handler.Request, prevModel *Model, currentModel *Model)
 	}
 
 	if response, err := client.Containers.Delete(context.Background(), projectID, containerID); err != nil {
-		if response.StatusCode == 409 {
-			// The deletion will fail if the there is an atlas cluster or network peering
-			// available in the same region a the container
-			return handler.ProgressEvent{
-				OperationStatus:  handler.Failed,
-				Message:          fmt.Sprintf("Please, make sure to delete the network peering and the atlas cluster before deleting the container: %s", err.Error()),
-				HandlerErrorCode: cloudformation.HandlerErrorCodeResourceConflict,
-			}, nil
-		}
-
-		return progressevents.GetFailedEventByResponse(fmt.Sprintf("Error getting resource: %s", err.Error()),
-			response.Response), nil
+		return errorHandling(client, response, err, projectID, containerID)
 	}
 
 	return handler.ProgressEvent{
 		OperationStatus: handler.Success,
 		Message:         "Delete Complete",
 	}, nil
 }
+
+func errorHandling(client *mongodbatlas.Client, response *mongodbatlas.Response, err error, projectID, containerID string) (handler.ProgressEvent, error) {
```

Comment:
Can we rename this method to `retryDeleteIfRequired()` or something? Right now it's not just handling errors but also tries to call client.Containers.Delete() again in certain case.
Ideally, we should try to extract retry logic out of this.